### PR TITLE
drivers/platform/stm32: stm32 pwm support.

### PR DESCRIPTION
**Pull request by pmallick-adi**
* Additionally the gpio pin for pwm requires to be configured as alternate functionality.

### CODE REVIEW

**Review by amiclaus — COMMENTED** [review]
I think this i a good moment to add a generic api for the pwm `no_os_pwm.c` here: https://github.com/analogdevicesinc/no-OS/tree/master/drivers/api and use the platform_ops approach.

**Comment by pmallick-adi** [comment]
> I think this i a good moment to add a generic api for the pwm `no_os_pwm.c` here: https://github.com/analogdevicesinc/no-OS/tree/master/drivers/api and use the platform_ops approach.

Sure then. Will add it.

**Comment by pmallick-adi** [comment]
Any update on this?

**Review by amiclaus — COMMENTED** [review]
please perform a rebase with the latest master branch

**Review by amiclaus — COMMENTED** [review]
there are some other platforms (aducm3029, axi_pwm) that also need update after the addition of platform ops.

**Comment by pmallick-adi** [comment]
Hi @amiclaus ,

No sure why the projects check is failing. Apart from that everything else looks good?

**Comment by pmallick-adi** [comment]
Hi @amiclaus ,

Looks good?

**Comment by buha** [comment]
@pmallick-adi ,
This looks good to me but there are projects that are still failing:
https://dev.azure.com/AnalogDevices/no-OS/_build/results?buildId=24852&view=logs&j=60d1ba4b-5db0-5461-607f-0c2a12524714&t=59f7d250-5a88-5157-974b-6c5f973eca2c

Can you see the output of this ?
You can download a log with the build errors and try to fix them.

**Comment by pmallick-adi** [comment]
> @pmallick-adi , This looks good to me but there are projects that are still failing: https://dev.azure.com/AnalogDevices/no-OS/_build/results?buildId=24852&view=logs&j=60d1ba4b-5db0-5461-607f-0c2a12524714&t=59f7d250-5a88-5157-974b-6c5f973eca2c
> 
> Can you see the output of this ? You can download a log with the build errors and try to fix them.

@buha I can see the build failure in the azure pipeline. However, I'm not sure how to access the build logs for each project failure.

**Comment by pmallick-adi** [comment]
Can we merge this now?

File: drivers/platform/aducm3029/aducm3029_pwm.h [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+/******************************************************************************/
+/*************************** Types Declarations *******************************/
+/******************************************************************************/
+/**
+ * @brief ADUCM3029 specific PWM platform ops structure
+ */
+extern const struct no_os_pwm_platform_ops aducm3029_pwm_ops;
+
+#endif // ADUCM3029_PWM_H_
```
Comment (amiclaus) [inline drivers/platform/aducm3029/aducm3029_pwm.h:59]: nit: new line

File: drivers/platform/mbed/mbed_pwm.h [hunk 2]
```diff
@@ -62,9 +62,10 @@ struct mbed_pwm_desc {
 	void *pwm_obj;		// Mbed PWM instance/object
 };
 
-/******************************************************************************/
-/************************ Public Declarations *********************************/
```
Comment (amiclaus) [inline drivers/platform/mbed/mbed_pwm.h:66]: unrelated change.

File: projects/iio_aducm3029/src.mk [hunk 3]
```diff
@@ -8,11 +8,13 @@ SRC_DIRS += $(PROJECT)/src \
...
 		$(NO-OS)/include
 SRCS += $(NO-OS)/drivers/api/no_os_irq.c
-SRCS += $(NO-OS)/drivers/api/no_os_uart.c
+SRCS += $(NO-OS)/drivers/api/no_os_uart.c 
+SRCS += $(NO-OS)/drivers/api/no_os_pwm.c
 
 ifeq '$(TINYIIOD)' 'y'
 SRC_DIRS += $(NO-OS)/iio/iio_app
 SRCS += $(NO-OS)/drivers/api/no_os_gpio.c \
+		$(NO-OS)/drivers/api/no_os_pwm.c \
```
Comment (amiclaus) [inline projects/iio_aducm3029/src.mk:17]: this seems to be added twice.

**Review by amiclaus — COMMENTED** [review]
some minor comments.

File: drivers/platform/mbed/mbed_pwm.h (hunk 2 above)
Comment (pmallick-adi) [reply drivers/platform/mbed/mbed_pwm.h:66]: This is not required anymore.

**Review by buha — APPROVED** [review]

**Review by amiclaus — APPROVED** [review]
